src/field.ml

```diff
@@ -39,9 +45,11 @@ module MakeBool (Size : SIZE) = struct
   type t = bool [@@deriving repr]
 
   let size = Size.size
+  (* you should raise an error is size is not 1 *)
 
   let set ~marker b ~off t =
     marker ();
+    (* any reason why you picked \254 and not \0 ? *)
```

Comment:
Yes, there are a lot of `\000` naturally because that's how bytes are initialised. In my experience, it made debugging easier to pick weirdly specific values. It makes apparent than a byte was never set, for instance.